mod/likes/start.php

```diff
@@ -222,3 +225,41 @@ function likes_count(ElggEntity $entity) {
 		return $entity->countAnnotations('likes');
 	}
 }
+
+/**
+ * Update annotation access to match that of the entity
+ *
+ * @param string     $event  'update:after'
+ * @param string     $type   'all'
+ * @param ElggEntity $entity The updated entity
+ * @return bool
+ *
+ * @access private
+ */
+function likes_access_sync($event, $type, $entity) {
+	if (!($entity instanceof \ElggEntity)) {
+		return true;
+	}
+	
+	// need to override access in case likes ended up with ACCESS_PRIVATE
+	// and to ensure write permissions
+	$ia = elgg_set_ignore_access(true);
+	$options = array(
+		'entity_guid' => $entity->getGUID(),
+                'annotation_name' => 'likes',
+		'wheres' => array(
+			"e.access_id != {$entity->access_id}"
```

Comment:
is the prefix of the annotations table `e`? Not `a`?

Reply:
elgg_get_annotations() is calling _elgg_services()->annotations->find() which is calling _elgg_get_metastring_based_objects() which uses e prefix as a generic prefix for both metadata and annotations tables

Reply:
Sorry for my mistake, forget my previous comment. You were right the table alias is wrong, it should be 'n_table' rather than 'e' !
I was too confident because I thought I tested it before my vacations, but it seems I was close to burn out... The fix doesn't work...
I changed the code and tested it this morning, it works now, I will push a new commit soon.